DUnitX.Exceptions.pas

```diff
@@ -0,0 +1,87 @@
+{***************************************************************************}
+{                                                                           }
+{           DUnitX                                                          }
+{                                                                           }
+{           Copyright (C) 2015 Vincent Parrett & Contributors               }
+{                                                                           }
+{           vincent@finalbuilder.com                                        }
+{           http://www.finalbuilder.com                                     }
+{                                                                           }
+{                                                                           }
+{***************************************************************************}
+{                                                                           }
+{  Licensed under the Apache License, Version 2.0 (the "License");          }
+{  you may not use this file except in compliance with the License.         }
+{  You may obtain a copy of the License at                                  }
+{                                                                           }
+{      http://www.apache.org/licenses/LICENSE-2.0                           }
+{                                                                           }
+{  Unless required by applicable law or agreed to in writing, software      }
+{  distributed under the License is distributed on an "AS IS" BASIS,        }
+{  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. }
+{  See the License for the specific language governing permissions and      }
+{  limitations under the License.                                           }
+{                                                                           }
+{***************************************************************************}
+
+unit DUnitX.Exceptions;
+
+interface
+
+{$I DUnitX.inc}
+
+uses
+  {$IFDEF USE_NS}
+  System.SysUtils,
+  {$ELSE}
+  SysUtils,
+  {$ENDIF}
+
  DUnitX.ComparableFormat;
+
+type
+  ETestFrameworkException = class(Exception);
+
+  ENotImplemented = class(ETestFrameworkException);
+
+  //base exception for any internal exceptions which cause the test to stop
+  EAbort = class(ETestFrameworkException);
+
+  ETestFailure = class(EAbort);
+
+  ETestFailureStrCompare = class(ETestFailure)
+  private
```

Comment:
This might need to be a EAbort exception descendant. From memory the exception types are tested against class types to figure out what should be done. ETestFailure has a particular path for failing the tests. If ETestFailureStrCompare is the same then there is no issue.